qt/widgets/common/src/ManageUserDirectories.cpp

```diff
@@ -9,30 +9,79 @@
 #include "MantidQtWidgets/Common/HelpWindow.h"
 #include <QDir>
 #include <QFileDialog>
+#include <QPointer>
 #include <QSettings>
 #include <QUrl>
 
+using Mantid::Kernel::ConfigService;
 using namespace MantidQt::API;
 
+namespace ButtonPrefix {
+const QString DATA{"pbData"};
+const QString SCRIPT{"pbScript"};
+const QString EXTENSIONS{"pbExt"};
+} // namespace ButtonPrefix
+
+namespace ConfigKeys {
+const std::string DATASEARCH_DIRS{"datasearch.directories"};
+const std::string PYTHONSCRIPTS_DIRS{"pythonscripts.directories"};
+const std::string USERPYTHONPLUGINS_DIRS{"user.python.plugins.directories"};
+const std::string DATASEARCH_ARCHIVE{"datasearch.searcharchive"};
+const std::string DEFAULT_FACILITY{"default.facility"};
+const std::string DEFAULTSAVE_DIR{"defaultsave.directory"};
```

Comment:
Would it be better if these sorts of variables were `static`?

Reply:
It doesn't make too much difference lifetime wise but I could move up the anonymous namespace so they are not visible outside of this code unit. I'll do that.